### ISSUE TITLE
Misleading documentation for hook parameters in estimators evaluate, predict, train methods

### ISSUE THREAD

**Issue by espinafre** [tensorflow/tensorflow#36516]
The documentation for pre-made estimators such as tensorflow.estimator.LinearRegressor says that each of the methods evaluate, predict and train take a parameter called "hooks", which is a list of tensorflow.train.SessionRunHooks. However, no such class exist in the v2.1 API; instead, the source code says that it expects instances of tensorflow.compat.v1.train.SessionRunHook.

In the example of LinearRegressor (other pre-made estimators share the same problem), this is the API doc, which I believe is generated directly from the source code (see the documentation of the parameters for the train, evaluate and predict methods):
https://www.tensorflow.org/api_docs/python/tf/estimator/LinearRegressor#evaluate
https://www.tensorflow.org/api_docs/python/tf/estimator/LinearRegressor#predict
https://www.tensorflow.org/api_docs/python/tf/estimator/LinearRegressor#train

This is the source code that says it expects a class from the compat.v1 API:
https://github.com/tensorflow/estimator/blob/a7ba3b45d07dd517a0e6ff38e90ae3aa240f424b/tensorflow_estimator/python/estimator/estimator.py#L1947

**Comment by Saduf2019** [comment]
@espinafre 
Could you please verify with the latest version if the issue still persist.

**Comment by sachinprasadhs** [comment]
Hi, Estimator will be deprecated from Tensorflow 2.12 since it uses Tensorflow V1 behavior.
Could you please migrate from Estimator to Keras API which will enable you to perform the aforementioned tasks, such as [model building](https://www.tensorflow.org/guide/keras/custom_layers_and_models), gradient application, [training](https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit), evaluation, and prediction.
Refer Migration guide here https://www.tensorflow.org/guide/migrate/migrating_estimator for more details. Thanks!
